@here/harp-mapview/lib/MapView.ts

```diff
@@ -1939,8 +1939,17 @@ export class MapView extends THREE.EventDispatcher {
 
     /**
      * Requests a redraw of the scene.
+     *
+     * @param forceRedraw If `true`, updates scene immediately, without waiting for the next
+     * animation frame. Note: Internal `maxFps` will be overriden as this inteded to be called
+     * from external render loop.
      */
-    update() {
+    update(forceUpdate?: boolean) {
```

Comment:
forceUpdate or forceRedraw?
I'm also wondering if having a parameter here is enough. If an external application is managing the rneder loop and requesting animation frames MapView should never request animation frames. Currently MapView will still request animation frames in a lot of situations (e.h. whenever data comes in, building animation is ongoing, ...).

So I think this needs to be a MapView option to completely switch to synchronous rendering.

Reply:
I think we can't rely on the user to deal with all `updates` that we have.
And this `forceUpdate` option is an easy solution for not to wait for the next animation frame and just re-render.

`forceUpdate` sounds better as we have `update` method.

Reply:
Why not? If we set synchronous rendering to true in the MapViewOptions, we could just do nothing in MapView.update, except triggering an event that an update was requested.
The application could register to that event and request an animation frame. In the animation frame the application would call MapView.redraw().
I don't think that the current solution will work if the application is drawing some additional stuff in the animation frame. If the animation frame is coming from map view that additional stuff would be missing, i.e. leading to some flickering.